.github/workflows/ai-revision.yaml

```diff
@@ -54,21 +61,50 @@ jobs:
           manubot_url=$(echo "$manubot_line" | awk -F"- " '{print $2}')
 
           pip install ${manubot_url}#egg=manubot[ai-rev]
+          
+          # install the multi-provider version of manubot-ai-editor over the current one, just for now
+          pip install git+https://github.com/manubot/manubot-ai-editor@langchain-anthropic-integration#egg=manubot-ai-editor
       - name: Revise manuscript
         env:
-          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
-          AI_EDITOR_LANGUAGE_MODEL: ${{ inputs.model }}
+          PROVIDER_API_KEY: ${{ secrets.PROVIDER_API_KEY || secrets.OPENAI_API_KEY }}
+          AI_EDITOR_MODEL_PROVIDER: ${{ inputs.model_provider }}
+          REPO_DEFAULT_MODEL_PROVIDER: ${{ vars.AI_EDITOR_MODEL_PROVIDER }}
+          AI_EDITOR_LANGUAGE_MODEL: ${{ inputs.model || vars.AI_EDITOR_LANGUAGE_MODEL }}
           AI_EDITOR_FILENAMES_TO_REVISE: ${{ inputs.file_names }}
           AI_EDITOR_CUSTOM_PROMPT: ${{ inputs.custom_prompt }}
           # More variables can be specified to control the behavior of the model:
           #  https://github.com/manubot/manubot-ai-editor/blob/main/libs/manubot_ai_editor/env_vars.py
-        run: manubot ai-revision --content-directory content/ --config-directory ci/
+        run: |
+          if [[ "${AI_EDITOR_MODEL_PROVIDER}" = "(repo default)" ]]; then
+            # attempt to retrieve repo default model provider from the AI_EDITOR_MODEL_PROVIDER repo var
+            if [[ ! -z "${REPO_DEFAULT_MODEL_PROVIDER}" ]]; then
+              echo "* Using repo-default model provider AI_EDITOR_MODEL_PROVIDER repo variable: ${REPO_DEFAULT_MODEL_PROVIDER}"
+              export AI_EDITOR_MODEL_PROVIDER="${REPO_DEFAULT_MODEL_PROVIDER}"
+            else
+              echo "Warning: Repo-default provider was specified, but no AI_EDITOR_MODEL_PROVIDER repo variable was set, defaulting to 'openai'"
+              export AI_EDITOR_MODEL_PROVIDER="openai"
+            fi
+          fi
```

Comment:
Do we have documentation about all these variables?

Reply:
We don't; I'll open a new PR in manubot-ai-editor to add that, thanks.